src/metrics/nom.rs

```diff
@@ -277,4 +499,87 @@ mod tests {
             ]
         );
     }
+
+    #[test]
+    fn javascript_unnamed_nom() {
+        check_metrics!(
+            "Ajax.getTransport = Try.these(
+                 function() {
+                     return function(){ return new XMLHttpRequest()}
+                 }
+             );",
+            "foo.js",
+            JavascriptParser,
+            nom,
+            [
+                (functions, 0, usize),
+                (closures, 2, usize),
+                (total, 2, usize)
+            ]
+        );
+    }
+
+    #[test]
+    fn javascript_arrow_nom() {
+        check_metrics!(
+            "var materials = [\"Hydrogen\"];
+             materials.map(material => material.length);
+             let add = (a, b)  => a + b;",
+            "foo.js",
+            JavascriptParser,
+            nom,
+            [
+                (functions, 1, usize), // add
+                (closures, 1, usize),  // materials.map
+                (total, 2, usize)
+            ]
+        );
+    }
+
+    #[test]
+    fn javascript_arrow_assignment_nom() {
+        check_metrics!(
+            "sink.onPull = () => { };",
```

Comment:
This one is kind of a corner case. I guess we'll never have a fully perfect way to differentiate between functions and unnamed functions

Reply:
So do you propose to delete that test?

Reply:
No, let's keep it, it was just a thought

Reply:
@marco-c every arrow function in the rhs of an assignment is named function or rhs of property definition `{foo: () => ...}`, or do you see any cases where it isn't the case ?

Reply:
@calixteman it is always the case, what I was thinking are cases like:
```
window.onload = function() { blabla }
```
which we consider a named function, because it theoretically is, but in practice it is an unnamed one if you look at it from the perspective of what the developer meant.